packages/types/src/auth/HttpSigner.ts

```diff
@@ -1,6 +1,20 @@
-import { HttpRequest } from "../http";
+import { HttpRequest, HttpResponse } from "../http";
 import { Identity } from "../identity/identity";
 
+/**
+ * @internal
+ */
+export interface ErrorHandler {
+  (signingProperties: Record<string, unknown>): <E extends Error>(error: E) => never;
```

Comment:
The error handler is expected to throw the error it receives, so it would be `never`.

I can expand it to `never | void`, but I think for now it's better to constrain the behavior to expect throwing the error.

I can add a line about it in the documentation.